lib-ee/emqx_ee_bridge/src/kafka/emqx_bridge_impl_kafka_consumer.erl

```diff
@@ -0,0 +1,308 @@
+    ConsumerConfig = [
+        {max_bytes, MaxBatchBytes},
+        {offset_reset_policy, OffsetResetPolicy}
+    ],
+    InitialState = #{
+        resource_id => emqx_bridge_resource:resource_id(kafka_consumer, BridgeName),
+        mqtt => #{
+            payload => MQTTPayload,
+            topic => MQTTTopic,
+            qos => MQTTQoS
+        },
+        kafka_topic => KafkaTopic
+    },
+    ClientID0 = emqx_bridge_impl_kafka:make_client_id(BridgeName),
+    ClientID = binary_to_atom(ClientID0),
+    ClientOpts0 =
+        case Auth of
+            none -> [];
+            Auth -> [{sasl, emqx_bridge_impl_kafka:sasl(Auth)}]
+        end,
+    ClientOpts = add_ssl_opts(ClientOpts0, SSL),
+    case brod:start_client(BootstrapHosts, ClientID, ClientOpts) of
+        ok ->
+            ?SLOG(info, #{
+                msg => "kafka_consumer_client_started",
+                instance_id => InstanceId,
+                kafka_hosts => BootstrapHosts
+            });
+        {error, Reason} ->
+            ?SLOG(error, #{
+                msg => "failed_to_start_kafka_consumer_client",
+                instance_id => InstanceId,
+                kafka_hosts => BootstrapHosts,
+                reason => Reason
+            }),
+            throw(failed_to_start_kafka_client)
+    end,
+    GroupID = ClientID0,
+    GroupSubscriberConfig =
+        #{
+            client => ClientID,
+            group_id => GroupID,
+            topics => [KafkaTopic],
+            cb_module => ?MODULE,
+            init_data => InitialState,
+            message_type => message,
+            consumer_config => ConsumerConfig,
+            group_config => GroupConfig
+        },
+    case start_pool(InstanceId, PoolSize, GroupSubscriberConfig) of
+        {ok, ChildrenIds} ->
+            ?tp(
+                kafka_consumer_start_pool_started,
+                #{
+                    pool => ChildrenIds,
+                    instance_id => InstanceId
+                }
+            ),
+            {ok, #{
+                children_ids => ChildrenIds,
+                kafka_client_id => ClientID
+            }};
+        {error, #{error := Reason2, started := Started}} ->
+            ?SLOG(error, #{
+                msg => "failed_to_start_kafka_consumer",
+                instance_id => InstanceId,
+                kafka_hosts => BootstrapHosts,
+                kafka_topic => KafkaTopic,
+                reason => Reason2
+            }),
+            stop_consumer_pool(Started),
+            stop_client(ClientID),
+            throw(failed_to_start_kafka_consumer)
+    end.
+
+-spec on_stop(manager_id(), state()) -> ok.
+on_stop(_InstanceID, State) ->
+    #{
+        children_ids := ChildrenIds,
+        kafka_client_id := ClientID
+    } = State,
+    stop_consumer_pool(ChildrenIds),
+    stop_client(ClientID),
+    ok.
+
+-spec on_get_status(manager_id(), state()) -> connected.
+on_get_status(_InstanceID, _State) ->
+    connected.
```

Comment:
- [ ] FIXME: should check if all consumers are connected or if client can access the leaders of all partitions (how to list partitions?).